src/mmrelay/meshtastic/connection.py

```
@@ -979,8 +1043,32 @@ def create_ble_interface(
+                        )
+                        if unresolved_teardown:
+                            facade.logger.warning(
+                                "Blocking BLE connect() for %s generation=%s due to "
+                                "unresolved teardown workers=%s",
+                                ble_address,
+                                connect_generation,
+                                unresolved_teardown,
+                            )
+                            raise TimeoutError(
+                                f"BLE teardown still unresolved for {ble_address}; waiting before connect()."
+                            )
```

Ruff (0.15.10)

[warning] Abstract `raise` to an inner function

(TRY301)